documentation/src/main/doc/modules/kafka/pages/inbound.adoc

```diff
@@ -137,6 +137,40 @@ The record written on the dead letter queue contains a set of additional headers
 * `dead-letter-partition` - the original partition of the record (integer mapped to String)
 * `dead-letter-offset` - the original offset of the record (long mapped to String)
 
+=== Handling deserialization failures
+
+Because deserialization happens before creating a `Message`, the failure strategy presented above cannot be applied.
+However, when a deserialization failure occurs, you can intercept it and provide a fallback value.
+If you don't, `null` will be used as fallback value.
+
+To achieve this, create a CDI bean implementing the {javadoc-base}/io/smallrye/reactive/messaging/kafka/DeserializationFailureHandler.html[DeserializationFailureHandler<T>] interface:
+
+[source, java]
+----
+@ApplicationScoped
+@Named("failure-fallback") // Set the name of the failure handler
+public static class MyDeserializationFailureHandler
+    implements DeserializationFailureHandler<JsonObject> { // Specify the expected type
+
+    @Override
+    public JsonObject handleDeserializationFailure(String topic, boolean isKey,
+            String deserializer, byte[] data,
+            Exception exception, Headers headers) {
+        return fallback;
+    }
+}
+----
+
+The bean must be exposed with the `@Named` qualifier specifying the name of the bean.
+Then, in the connector configuration, specify the following attribute:
+
+* `mp.messaging.incoming.$channel.value-deserialization-failure-handler`: name of the bean handling deserialization failures happening for the record's value,
```

Comment:
Not huge, but maybe switch the ordering of the config keys to have the "key" first as it's mentioned first in other situations

Reply:
Good point!